Formula/interactive-rebase-tool.rb

```diff
@@ -0,0 +1,38 @@
+class InteractiveRebaseTool < Formula
+  desc "Ncurses sequence editor for git interactive rebase"
+  homepage "https://gitrebasetool.mitmaro.ca/"
+  url "https://github.com/MitMaro/git-interactive-rebase-tool/archive/0.7.0.tar.gz"
+  sha256 "08e5d6dd9beacf7806abd74edfa9e7654ccb2ffc083b2fd8617d132951eee5bd"
+
+  depends_on "rust" => :build
+  depends_on "openssl"
+
+  def install
+    system "cargo", "install", "--root", prefix, "--path", "."
+    ohai "Remember to configure git to use interactive-rebase-tool."
+    ohai 'Execute "git config --global sequence.editor interactive-rebase-tool" to use interactive-rebase-tool by default.'
+  end
+
+  test do
+    require "pty" # required for interactivity
+
+    # create a git repository
+    mkdir "repo" do
+      system "git", "init", "."
+
+      3.times do |i|
+        system "echo #{i} >> file#{i}.out"
+        system "git", "add", "file#{i}.out"
+        system "git", "commit", "-m", "commit #{i}.out"
+      end
+
+      # set interactive-rebase-tool as git's sequence editor
+      system "git", "config", "sequence.editor", "interactive-rebase-tool"
+
+      # execute interactive git rebase
+      PTY.spawn("cd $HOME/repo && git rebase -i HEAD~2") do |_input, output, _pid|
+        output.puts "W" # send uppercase W to interactive-rebase-tool to execute the rebase
+      end
```

Comment:
Can we check that the rebase was done?

Reply:
It's a bit difficult for me to reason about, given the app is interactive. Regardless, the app should return its status code, so I assume if it returns 0, it did its job (which is to write out the rebase todo file) correctly. Perhaps an arbitrary rebase won't complete correctly, but that's not necessarily the fault of `interactive-rebase-tool`.

Reply:
Hey @fxcoudert and @jay-hankins ! I'm the author of the interactive-rebase-tool and would love to see this included in the homebrew core! I haven't had much experience with creating homebew packages, but I will gladly assist @jay-hankins in getting this in!

The rebase tool does not interact directly with git and instead modifies the git rebase todo file that is generally located in `.git/rebase-merge/git-rebase-todo`. A simple check would be to ensure that `interactive-rebase-tool --version` returns a non-zero status code.

I also think a great test would be to check that a sample rebase file is updated appropriately after a certain actions have occurred? That way the tool could be tested without the need to use `git`. This would be similar to how the test for vim is performed: https://github.com/Homebrew/homebrew-core/blob/master/Formula/vim.rb

Reply:
Hi @MitMaro, great tool! Long term solution (new feature) would be to implement non-interactive mode, so we would be able to send a command like `drop` for example and compare files before and after.

Reply:
In the meantime, we can work out a test that checks the `.git/rebase-merge/git-rebase-todo` file for an expected result. 

@MitMaro I'm interested if you've got an idea here for how to create the git-rebase-todo file, and have interactive-rebase-tool modify it, in a repeated, automated fashion. Right now as I see it, we can't do a simple diff assertion because the commit hashes would differ. You did mention a way to do testing without git; I'm not sure that's necessary as anyone downloading this tool would have git and expect it to work in that flow. However, the test itself might be easier to accomplish if we can generate a git-rebase-todo file for it to manipulate.

@igas Interested in ideas you have as well. Clearly the cleanest solution is a non-interactive mode for the tool, but I hope we can get it into homebrew even before that.

Reply:
I think this is a good way to test it. I would send drop or squash that changes the history and do `git log` after to assert changes been made.